### PR TITLE
Adding nightly validation to autogen.sh. 

### PR DESCRIPTION
**Pull request by DavidDeSimone**
We have had a number of new contributors not realize that remacs requires nightly rust, and this has caused problems. This should make the build fail fast if nightly is not installed.

### CODE REVIEW

File: autogen.sh [hunk 1]
```diff
@@ -130,6 +133,15 @@ case $do_autoconf,$do_git in
     do_autoconf=true;;
 esac
 
+if $require_nightly; then
+    echo "Validation Rust install ..."
```
Comment (birkenfeld) [inline autogen.sh]: "Validating"?

File: autogen.sh [hunk 2]
```diff
@@ -130,6 +133,15 @@ case $do_autoconf,$do_git in
...
+    command -v rustc >/dev/null 2>&1 || { echo >&2 "Remacs requires rust to be installed in order to build. Please install it via rustup: https://www.rustup.rs/; Aborting."; exit 1; }
+    command -v cargo >/dev/null 2>&1 || { echo >&2 "I require foo but it's not installed.  Aborting.Remacs requires cargo to be installed in order to build. Please install it via rustup: https://www.rustup.rs/; Aborting."; exit 1; }
```
Comment (birkenfeld) [inline autogen.sh]: message seems jumbled?

File: autogen.sh [hunk 3]
```diff
@@ -130,6 +133,15 @@ case $do_autoconf,$do_git in
...
+
+    rustc --version | grep nightly || { echo >&2 "Remacs currently requires nightly Rust. If you do not have nightly Rust, you should install rustup at https://www.rustup.rs/, and run 'rustup install nightly && rustup default nightly"; exit 1; }
```
Comment (birkenfeld) [inline autogen.sh]: I'd suggest `rustup override ...` instead, to make the nightly choice only for Remacs.

File: autogen.sh (hunk 2 above)
Comment (DavidDeSimone) [inline autogen.sh]: Whoops, bade copy/paste, let me fix that.

File: autogen.sh [hunk 4]
```diff
@@ -130,6 +134,15 @@ case $do_autoconf,$do_git in
     do_autoconf=true;;
 esac
 
+if $require_nightly; then
+    echo "Validating Rust install ..."
+    command -v rustc >/dev/null 2>&1 || { echo >&2 "Remacs requires rust to be installed in order to build. Please install it via rustup: https://www.rustup.rs/; Aborting."; exit 1; }
```
Comment (birkenfeld) [inline autogen.sh]: hmm, actually these checks are not depending on nightly/stable, should they be moved outside the `if`?

File: autogen.sh [hunk 5]
```diff
@@ -130,6 +134,15 @@ case $do_autoconf,$do_git in
...
+    command -v cargo >/dev/null 2>&1 || { echo >&2 "Remacs requires cargo to be installed in order to build. Please install it via rustup: https://www.rustup.rs/; Aborting."; exit 1; }
+
+    rustc --version | grep nightly || { echo >&2 "Remacs currently requires nightly Rust. If you do not have nightly Rust, you should install rustup at https://www.rustup.rs/, and run 'rustup install nightly && rustup override nightly' in the remacs directory."; exit 1; }
```
Comment (birkenfeld) [inline autogen.sh]: it's `rustup override set nightly` I think.

Comment (DavidDeSimone) [inline autogen.sh]: you are correct, updating

File: autogen.sh (hunk 4 above)
Comment (DavidDeSimone) [inline autogen.sh]: good callout

**Comment by Wilfred** [comment]
Excellent, thanks :)

**Pull request merged**